### ISSUE TITLE
Training scheme modifications for small GPUs

### ISSUE THREAD

**Issue by justanhduc** [tmp-iclr/convmixer#13]
Hi authors. Your paper has demonstrated a quite intriguing observation. I wish you luck with your submission.
Thanks for sharing the code of the submission. When running the code, I got an issue regarding OOM when using the default batch size of 64. In the end I can only run with 8 samples per batch per GPU as my GPUs have only 11GB. I would like to know if you have tried smaller GPUs and achieved the same results. So far, besides learning rate modified according to the linear rule, I haven't made any change yet. If you tried training using smaller GPUs before, could you please share your experience? Thank you very much!

**Comment by tmp-iclr** [comment]
Thanks!

We trained the ConvMixers in this repo with 10x RTX8000s which have 48GB of memory each. It would indeed be challenging to train our largest models on, e.g., 2080Tis with only 11GB each (so no, we have not achieved the same results with smaller GPUs).

ConvMixers are unfortunately quite compute- and memory-hungry with small patch sizes, which we used for most of the results in the paper (patch_size=7). We think this is largely due to the fact that large-kernel depthwise convolution is poorly-optimized, which we are looking into improving. 

I recommend trying to train ConvMixers with a larger patch size, like patch_size=14. This significantly decreases the compute and memory requirements in exchange for some accuracy. We are optimistic that parameter tuning could result in better accuracy for large patch sizes than what we presented in the paper.

Let me know if you have any other questions or comments.

**Comment by tmp-iclr** [comment]
We just happened to not try that experiment due to limited availability of compute. However, we actually ran that exact experiment recently (ConvMixer-1536/20 with the _only change_ being patch_size=14, no changes to training setup). It achieved 78.92% top-1 accuracy and trained much more quickly -- in around 3 days on 10x RTX8000s.

**Comment by tmp-iclr** [comment]
I'm going to close this issue, but feel free to reopen it or open a new issue if you have more questions or comments.
